Assets/Scripts/ProjectileHazardScript.cs

```diff
@@ -15,25 +16,25 @@ public class ProjectileHazardScript : MonoBehaviour
     void Start()
     {
         isMoving = false;
+        playerRocket = GameObject.Find("Rocket");
     }
 
     // Update is called once per frame
-    void Update()
+    private void Update()
     {
         Collider[] hitColliders = Physics.OverlapSphere(transform.position,pullRadius);
         
         foreach (var currentCollider in hitColliders)
         {
-            if (!isMoving){
+            if (isMoving){
+                GetComponent<Rigidbody>().AddForce(initialRocketPosition * strengthOfAttraction * Time.deltaTime);                
+            } else
                 if (currentCollider.tag == "Player")
                 {
                     initialRocketPosition = playerRocket.transform.position - transform.position;
```

Comment:
`targetVector = playerRocket.transform.position - transform.position;`

What you are doing here is you are calculating the vector the projectile needs to travel along in order to hit the player if they were to stop moving at the time this statement is evaluated. transform.position (and subsequently playerRocket.transform.position) is a vector. What this statement does is set the targetVector equal to a displacement vector in the direction of the rocket.